File: skills/astra-toolkit/clients/typescript/examples/collections-insert-many-vectorize.ts
import {
  DataAPIClient,
  CollectionInsertManyError,
} from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db(process.env.ASTRA_DB_API_ENDPOINT!, {
  token: process.env.ASTRA_DB_APPLICATION_TOKEN!,
});
const collection = database.collection("**COLLECTION_NAME**");

// Insert documents into the collection
(async function () {
  try {
    const result = await collection.insertMany([
      {
        name: "Jane Doe",
        age: 42,
        $vectorize: "Text to vectorize for this document",
      },
      {
        nickname: "Bobby",
        $vectorize: "Text to vectorize for this document",
      },
    ]);
  } catch (error) {
    if (error instanceof CollectionInsertManyError) {
      console.log(error.insertedIds());
    }
  }
})();

File: skills/astra-toolkit/clients/go/examples/collections-insert-many-vectorize.go
package main

import (
	"context"
	"log"
	"os"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()
	// Get an existing collection
	client := astra.NewClient()

	database := client.Database(
		os.Getenv("ASTRA_DB_API_ENDPOINT"),
		options.API().SetToken(os.Getenv("ASTRA_DB_APPLICATION_TOKEN")),
	)

	collection := database.Collection("**COLLECTION_NAME**")

	// Insert documents into the collection
	_, err := collection.InsertMany(
		ctx,
		[]map[string]any{
			{
				"name":       "Jane Doe",
				"age":        42,
				"$vectorize": "Text to vectorize for this document",
			},
			{
				"nickname":   "Bobby",
				"$vectorize": "Text to vectorize for this document",
			},
		},
	)
	if err != nil {
		log.Fatal(err)
	}
}
